### PR TITLE
Send payload id if applicable when an exception is caught

### PR DESCRIPTION
**Pull request by o0Ignition0o**
If an exception is caught during planning, we previously didn't send the messageId.

The response could thus not be properly routed back to the proper sender, and would lead to such errors as:

```
value: TypeError: Error parsing args at position 1: missing field `id`
```

This changeset adds the messageId if possible.

### CODE REVIEW

File: router-bridge/js-src/plan_worker.ts [hunk 1]
```diff
@@ -245,8 +245,14 @@ const updateQueryPlanner = (
 
 async function run() {
   while (true) {
+    // If an Exception is raised,
+    // there won't be any messageId to send a payload back to the router,
+    // unless we keep it in this scope.
+    let messageId;
     try {
+      messageId = "";
```
Comment (goto-bus-stop) [inline router-bridge/js-src/plan_worker.ts:253]:
```suggestion
    let messageId = "";
    try {
```

Comment (goto-bus-stop) [reply router-bridge/js-src/plan_worker.ts:253]: same thing but more compact

**Review by goto-bus-stop — COMMENTED** [review]
Hmm. <strike>It looks like this could only occur if `receive()` errored? I'm not clear on when that would happen (though this patch would surely help to figure that out 😄 )</strike>

I wonder if we should accept `Option<String>` on the rust side rather than using a default empty value? Or is that super invasive?

**Review by goto-bus-stop — APPROVED** [review]
Ohh wait. I reread it after. My comment about `receive()` is incorrect.

This patch looks good, we would only get the default empty value if `receive()` fails, which should not happen ( 😇 ), and we now get the correct value if QP fails, which we previously didn't.

Adjusting my question: is getting an empty value going to be a problem in the router? It shouldn't happen, but it might ;p Either way this is better than before.

**Comment by duckki** [comment]
This seems working. I don't see this error with this PR.

**Pull request merged**